### PR TITLE
Add discard comments popup and cancel cta in edit expenses

### PR DESCRIPTION
**Pull request by yksolanki9**
Discard comments popup:
https://user-images.githubusercontent.com/39493102/131531709-5be2f253-5b04-4c9b-9f04-24005216ea83.mp4

Cancel cta in edit expenses:
https://user-images.githubusercontent.com/39493102/131531808-b581955e-68ea-464a-b91e-9a9e1daff912.mp4

### CODE REVIEW

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.scss [hunk 1]
```diff
@@ -45,6 +45,11 @@
     }
   }
 
+  &--cta-container {
+    padding: 14px 16px 14px 16px;
```
Comment (Dimple16) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.scss:49]: padding: 14px 16px;

File: src/app/shared/components/comments-history/view-comment/view-comment.component.ts [hunk 2]
```diff
@@ -80,13 +82,40 @@ export class ViewCommentComponent implements OnInit {
...
+          },
+        },
+        cssClass: 'pop-up-in-center',
+      });
+
+      await unsavedChangesPopOver.present();
+      const { data } = await unsavedChangesPopOver.onWillDismiss();
+
+      if (data && data.action === 'discard') {
+        this.trackingService.viewComment({ Asset: 'Mobile' });
```
Comment (Dimple16) [inline src/app/shared/components/comments-history/view-comment/view-comment.component.ts:108]: remove `Asset: 'Mobile'`

File: src/app/shared/components/comments-history/view-comment/view-comment.component.ts [hunk 3]
```diff
@@ -80,13 +82,40 @@ export class ViewCommentComponent implements OnInit {
...
+        this.modalController.dismiss();
+      }
     } else {
-      this.trackingService.viewComment({ Asset: 'Mobile' });
-      this.modalController.dismiss();
+      if (this.isCommentAdded) {
+        this.trackingService.addComment({ Asset: 'Mobile' });
```
Comment (Dimple16) [inline src/app/shared/components/comments-history/view-comment/view-comment.component.ts:113]: same as above

**Review by Dimple16 — CHANGES_REQUESTED** [review]
few comments

File: src/app/shared/components/comments-history/view-comment/view-comment.component.ts (hunk 2 above)
Comment (yksolanki9) [reply src/app/shared/components/comments-history/view-comment/view-comment.component.ts:108]: These change have been taken care of in another pr

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**